### PR TITLE
Fix incompatibility with Woodstox StAX implementation

### LINKED ISSUE #105: XML output may not be well formed if a third party XMLOutputFactory is used
Original issue reported by @syaninas:

- https://github.com/junit-team/junit5/issues/3715

> JUnit version: 5.10.2
> JUnit platform version: 1.10.2
> 
> I am trying to use the Open Test Reporting XML format as described [here](https://junit.org/junit5/docs/current/user-guide/#junit-platform-reporting-open-test-reporting). But the XML file produced is always malformed. The file begins with two newlines before the XML declaration, which isn't allowed.
> 
> I have tracked the issue down to two items:
> 
>     I was using [Woodstox](https://github.com/FasterXML/woodstox) as my XML implementation provider, and the associated jars were on my classpath
>     [open-test-reporting](https://github.com/ota4j-team/open-test-reporting)'s [DefaultDocumentWriter](https://github.com/ota4j-team/open-test-reporting/blob/main/events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java?rgh-link-date=2024-03-05T18%3A08%3A46Z) direct insertion of newlines to the bufferedWriter backing the XMLStreamWriter.
> 
> This direct newline insertion reaches around any internal buffering the XMLStreamWriter may be doing. Woodstox won't write to the backing file until enough XML content is written to fill its internal buffers, but the newlines are written to file stream directly. This results in the newlines added "after" the XML declaration in DefaultDocumentWriter and "before" the infrastructure report element appended by OpenTestReportGeneratingListener to actually appear in the file before any XML content.
> 
> I initially encountered the issue while running JUnit from the command line inside a much larger project with many dependencies (including Woodstox). Removing the Woodstox dependency is not realistic. A workaround via setting javax.xml.stream.XMLOutputFactory system property to com.sun.xml.internal.stream.XMLOutputFactoryImpl does resolve the issue, but is not ideal.
> 
> I have opened the issue on JUnit rather than open-test-reporting since JUnit is much more active, the reproduction is entirely JUnit driven, and in case there is some larger change to be made.
> Steps to reproduce
> 
> I have created a minimal example project which exhibits this issue:
> [syaninas/opentestreportingxmlbug](https://github.com/syaninas/opentestreportingxmlbug)
> 
> Inside that project as-is running ./gradlew test will result in a junit-platform-events-<random-id>.xml file written directly into the repo root that is malformed as described above.
> Context
> 
>     Used versions (Jupiter/Vintage/Platform):
>     JUnit version: 5.10.2
>     JUnit platform version: 1.10.2
> 
>     Build Tool/IDE:
>     Gradle 8.4

### PR DESCRIPTION
**Pull request by marcphilipp**
- **Add separate test task to test against Woodstox StAX implementation**
- **Flush writers before/after appending line breaks**

Fixes #105.

### CODE REVIEW

File: events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java
Comment (mmerdes) [inline events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java:1]: The new method seems straightforward enough, so maybe it doesn't warrant its own unit test. But have you verifying the changed behaviour with some form of test - maybe mocking the writers?

**Review by mmerdes — APPROVED** [review]
Looks plausible to me - see some minor comments below. 
Cannot judge the gradle part so well - but then you are the expert here ;)

File: events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java
Comment (marcphilipp) [reply events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java:1]: I've added a separate Gradle `Test` task that uses the Woodstox StAX implementation. The existing tests failed when executed using that modified runtime classpath. I think that should be sufficient.

File: events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java [hunk 1]
```diff
@@ -135,6 +135,12 @@ public void close() throws IOException {
 		}
 	}
 
+	private void newLine() throws IOException, XMLStreamException {
```
[the remark this answers is not in the document]
Comment (marcphilipp) [reply events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java]: Renamed in 7cac517.

File: events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java
Comment (mmerdes) [reply events/src/main/java/org/opentest4j/reporting/events/api/DefaultDocumentWriter.java:1]: Yes, if the existing tests did fail at the beginning, this should suffice.

**Pull request merged**